src/nvim/lua/executor.c

```diff
@@ -870,6 +881,49 @@ static int nlua_print(lua_State *const lstate)
   return lua_error(lstate);
 }
 
+/// require used for --startuptime
+///
+/// @param  lstate  Lua interpreter state.
+static int nlua_require(lua_State *const lstate)
+  FUNC_ATTR_NONNULL_ALL
+{
+  const char *name = luaL_checkstring(lstate, 1);
+
+  // try cached module from package.loaded first
+  // NOTE: _LOADED is used in PUC and LuaJIT, but
+  //       it might be implementation defined?
+  lua_getfield(lstate, LUA_REGISTRYINDEX, "_LOADED");
+  lua_getfield(lstate, 2, name);
+  if (lua_toboolean(lstate, -1)) {
+    return 1;
+  }
+  lua_pop(lstate, 2);
+
+  proftime_T rel_time;
+  proftime_T start_time;
+  FILE * const l_time_fd = time_fd;
+
+  // push previous require and module name
+  lua_rawgeti(lstate, LUA_REGISTRYINDEX, require_ref);
+  lua_pushvalue(lstate, 1);
+
+  // call it directly if log file is closed
+  if (l_time_fd == NULL) {
+    lua_call(lstate, 1, 1);
+    return 1;
+  }
+
+  time_push(&rel_time, &start_time);
+  int res = lua_pcall(lstate, 1, 1, 0);
+  if (res == 0) {
+    vim_snprintf((char *)IObuff, IOSIZE, "require %s", name);
```

Comment:
nit: would require(foo) be a bit more recognizable/intuitive in the log?
```suggestion
    vim_snprintf((char *)IObuff, IOSIZE, "require('%s')", name);
```

Reply:
(would match our style guide ;))

Reply:
Not sure, but `--startuptime` is used by some plugins, and as it is right now is maybe easier to parse?

Reply:
But looking at the output now, I would argue for a change in the other direction:

`requiring %s`

(as the corresponding vimscript output is `sourcing <file>`, `opening buffers`, etc.)

Reply:
As this will be a breaking change for startuptime plugins (@tweekmonster @dstein64 ) anyway, should we take the opportunity to do that? I don't care either way, but the current inconsistency (in this PR) bugs me.

Reply:
No objection to exploring that, but let's leave it for a separate PR?